services/src/main/scala/cromwell/services/metadata/impl/MetadataDatabaseAccess.scala

```diff
@@ -304,4 +305,23 @@ trait MetadataDatabaseAccess {
       queryResults <- summariesToQueryResults(workflows)
     } yield (WorkflowQueryResponse(queryResults, count), queryMetadata(count))
   }
+
+  def deleteNonLabelMetadataEntriesForWorkflow(rootWorkflowId: WorkflowId)(implicit ec: ExecutionContext): Future[Int] = {
+    import cromwell.core.WorkflowState
+
+    ((metadataDatabaseInterface.isRootWorkflow(rootWorkflowId.toString), metadataDatabaseInterface.getWorkflowStatus(rootWorkflowId.toString)) mapN {
+      case (None, _) =>
+        Future.failed(new Exception(s"""Metadata deletion precondition failed: workflow ID "$rootWorkflowId" not found in summary table"""))
+      case (Some(false), _) =>
+        Future.failed(new Exception(s"""Metadata deletion precondition failed: workflow ID "$rootWorkflowId" is not a root workflow"""))
+      case (_, None) =>
+        Future.failed(new Exception(s"""Metadata deletion precondition failed: workflow ID "$rootWorkflowId" did not have a status in the summary table"""))
+      case (Some(true), Some(status)) =>
+        if (WorkflowState.WorkflowStateValues.filter(_.isTerminal).map(_.toString).contains(status))
+          metadataDatabaseInterface.deleteNonLabelMetadataForWorkflow(rootWorkflowId.toString)
+        else
+          Future.failed(new Exception(s"""Metadata deletion precondition failed: workflow ID "$rootWorkflowId" was in non-terminal status "$status""""))
```

Comment:
Squawking Out Loud:
In `WorkflowState`:

```
  lazy val TerminalWorkflowStateValues = WorkflowState.WorkflowStateValues.filter(_.isTerminal)

  def isTerminal(stringState: String): Boolean = {
    WorkflowState.TerminalWorkflowStateValues.map(_.toString).contains(stringState)
  }
```

Reply:
Here:
```
      case (Some(true), Some(state)) if WorkflowState.isTerminal(state) =>
        metadataDatabaseInterface.deleteNonLabelMetadataForWorkflow(rootWorkflowId.toString)
      case (Some(true), Some(state)) =>
        Future.failed(new Exception(s"""Metadata deletion precondition failed: workflow ID "$rootWorkflowId" was in non-terminal status "$state""""))
```

Reply:
Looking at `WorkflowState` gave me an even better idea:
```
WorkflowState.withName(status).isTerminal
```